src/ucp/wireup/wireup.c

```diff
@@ -762,49 +742,93 @@ ucp_wireup_connect_lane(ucp_ep_h ep, unsigned ep_init_flags,
-            ep->uct_eps[lane] = uct_ep;
-        } else {
-            uct_ep = ep->uct_eps[lane];
-            ucs_assert(ucp_wireup_ep_test(uct_ep));
+    if (ep->uct_eps[lane] == NULL) {
+        status = ucp_wireup_ep_create(ep, &uct_ep);
+        if (status != UCS_OK) {
+            /* coverity[leaked_storage] */
+            return status;
         }
 
-        if (!(ep_init_flags & (UCP_EP_INIT_CM_WIREUP_CLIENT))) {
-            ucs_trace("ep %p: connect uct_ep[%d]=%p to addr[%d] wireup", ep,
-                      lane, uct_ep, addr_index);
-            connect_aux = !ucp_ep_init_flags_has_cm(ep_init_flags) &&
-                          (lane == ucp_ep_get_wireup_msg_lane(ep));
-            status = ucp_wireup_ep_connect(ep->uct_eps[lane], ep_init_flags,
-                                           rsc_index,
-                                           ucp_ep_get_path_index(ep, lane),
-                                           connect_aux, remote_address);
-            if (status != UCS_OK) {
-                return status;
-            }
+        ucs_trace("ep %p: assign uct_ep[%d]=%p wireup", ep, lane, uct_ep);
+        ep->uct_eps[lane] = uct_ep;
+    } else {
+        uct_ep = ep->uct_eps[lane];
+        ucs_assert(ucp_wireup_ep_test(uct_ep));
+    }
+
+    if (!(ep_init_flags & UCP_EP_INIT_CM_WIREUP_CLIENT)) {
+        ucs_trace("ep %p: connect uct_ep[%d]=%p to remote addr %p wireup", ep,
+                  lane, uct_ep, remote_address);
+        connect_aux = !ucp_ep_init_flags_has_cm(ep_init_flags) &&
+                      (lane == ucp_ep_get_wireup_msg_lane(ep));
+        status = ucp_wireup_ep_connect(ep->uct_eps[lane], ep_init_flags,
+                                       rsc_index, path_index, connect_aux,
+                                       remote_address);
+        if (status != UCS_OK) {
+            return status;
         }
+    }
 
-        ucp_worker_iface_progress_ep(wiface);
+    ucp_worker_iface_progress_ep(wiface);
 
-        return UCS_OK;
-    }
+    return UCS_OK;
+}
 
-    return UCS_ERR_UNREACHABLE;
+ucs_status_t
+ucp_wireup_connect_lane(ucp_ep_h ep, unsigned ep_init_flags,
+                        ucp_lane_index_t lane, unsigned path_index,
+                        const ucp_unpacked_address_t *remote_address,
+                        unsigned addr_index)
+{
+    ucp_worker_h worker = ep->worker;
+    ucp_rsc_index_t rsc_index;
+    ucp_worker_iface_t *wiface;
+    ucp_address_entry_t *address;
+
+    ucs_trace("ep %p: connect lane[%d]", ep, lane);
+
+    ucs_assert(lane != ucp_ep_get_cm_lane(ep));
+
+    ucs_assert_always(remote_address != NULL);
+    ucs_assert_always(remote_address->address_list != NULL);
+    ucs_assert_always(addr_index <= remote_address->address_count);
+
+    rsc_index  = ucp_ep_get_rsc_index(ep, lane);
+    wiface     = ucp_worker_iface(worker, rsc_index);
+    address    = &remote_address->address_list[addr_index];
```

Comment:
can we move this to right before line 827? or remove the local variable 'address'?